h2o-py/tests/testdir_algos/xgboost/pyunit_PUBDEV_6117_xgboost_compare.py

```diff
@@ -39,7 +39,7 @@ def comparison_test():
                                 dtrain=nativeXGBoostInput, num_boost_round=nativeXGBoostParam[1])
         nativePred = nativeModel.predict(data=nativeXGBoostInput, ntree_limit=nativeXGBoostParam[1])
         pyunit_utils.summarizeResult_binomial(h2oPredictD, nativePred, -1, -1, -1,
-                                              -1, tolerance=1e-10)
+                                              -1, tolerance=1e-6)
```

Comment:
@tomasfryda  this is blocker for me. XGBoost is designed to match native. If it doesn't I need to know the reason. I tried image dev-python-3.7 which has xgboost==1.5.0 and test passed. I upgraded to xgboost==1.6.2 and the test passed, so the problem must be different from xgboost version.. know precision issues was introduced in xgboost==1.3. I will continue the investigation

Reply:
Thank you for notifying me. 

This is what I get locally from the current master:
```
[17:16:44] WARNING: /Users/travis/build/dmlc/xgboost/src/learner.cc:541: 
Parameters: { nround, silent } might not be used.

  This may not be accurate due to some parameters are only used in language bindings but
  passed down to XGBoost core.  Or some parameters are not used but slip through this
  verification. Please open an issue if you find above cases.


[17:16:44] WARNING: /Users/travis/build/dmlc/xgboost/src/learner.cc:1061: Starting in XGBoost 1.3.0, the default evaluation metric used with the objective 'binary:logistic' was changed from 'error' to 'logloss'. Explicitly set eval_metric if you'd like to restore the old behavior.
H2OXGBoost train time is -0.001s.  Native XGBoost train time is -1s.
  H2OXGBoost scoring time is -1s.  Native XGBoost scoring time is -1s.
Export File progress: |██████████████████████████████████████████████████████████████████████████████ (done)| 100%
100% |██████████████████████████████████████████████████| Reading data [done]
Traceback (most recent call last):
  File "tests/testdir_algos/xgboost/pyunit_PUBDEV_6117_xgboost_compare.py", line 47, in <module>
    pyunit_utils.standalone_test(comparison_test)
  File "/Users/tomasfryda/sources/h2o-3/h2o-py/tests/pyunit_utils/utilsPY.py", line 694, in standalone_test
    test()
  File "tests/testdir_algos/xgboost/pyunit_PUBDEV_6117_xgboost_compare.py", line 42, in comparison_test
    -1, tolerance=1e-10)
  File "/Users/tomasfryda/sources/h2o-3/h2o-py/tests/pyunit_utils/utilsPY.py", line 4071, in summarizeResult_binomial
    "very different.".format(h2oPredictLocalD[colnames[2]][ind], nativePred[ind])
AssertionError: H2O prediction prob: 0.022199217230081558 and native XGBoost prediction prob: 0.02219921536743641.  They are very different.
Closing connection _sid_8f1f at exit
H2O session _sid_8f1f closed.
```
When I increase the tolerance to 1e-8 it still fails.
```
AssertionError: H2O prediction prob: 0.1312842220067978 and native XGBoost prediction prob: 0.13128423690795898.  They are very different.
```

But with 1e-7 it seems to work locally but it sometimes failed on jenkins.

Reply:
I tried also dev-python-3.7:43 and it fails there, this image is used only for performance tests, so we didn't know it before. I will check who is actually giving the different output. If h2o or xgboost.